lib/migration-chunks/validation/field.js

```diff
@@ -153,9 +168,38 @@ const checks = {
         step
       ));
     }
+  },
+
+  'field/rename': (errors, step, validationContext) => {
+    const oldId = step.payload.fieldId;
+    const newId = step.payload.props.newId;
+
+    if (oldId === newId) {
+      errors.push(invalidActionError(
+        fieldErrors.changeId.ID_MUST_BE_DIFFERENT(oldId),
+        step
+      ));
+    } else if (fieldExists(validationContext, newId)) {
+      errors.push(invalidActionError(
+        fieldErrors.changeId.ID_ALREADY_EXISTS(
+          oldId,
+          newId,
+          step.payload.contentTypeId
+        ),
+        step
+      ));
+    }
+    // if the id is switched back to something changed before, we forget that id
```

Comment:
Shouldn't this add an error, as we rename the same field twice in one migration?

Reply:
No, it could be a different field on the same content type that is getting the id that another field previously had (like here https://github.com/contentful/migration-cli/blob/5ebc8db380f937ce76f8e9f593bff73775fabec5/test/unit/lib/migration-chunks/validation/field/change-id.spec.js#L59)

Reply:
I think I see your point, but in this example you actually do rename the same field twice 😆 

So if we rename `a -> b` and `c -> a` we forget about the `a -> b` renaming?

Reply:
Yes. Because if someone then tries to update `a`, that is now possible and we no longer want to tell them "hey, you can't do that because a has been renamed"